### ISSUE TITLE
docker cp - high memory ussage

### ISSUE THREAD

**Issue by vmindru** [moby/moby#30365]
When doing docker cp container:path /local_path docker  of a large file docker will consume approximative 
local_path file size * 2. 

```
[root@vmutil /]# docker --version
Docker version 1.12.5, build 047e51b/1.12.5
[root@vmutil /]#
```

reproduce

```
docker exec big_pare dd if=/dev/urandom of=/root/file bs=10M count=100
docker cp big_pare:/root/file ./file4
```
observer high docker memorry usage
```
[vmindru@vmutil docker-cp]$ top -d 0.5 -n 1 -b | grep docker                                              
20736 root      20   0 3406456 2.170g   6080 S   0.0 78.1   0:04.38 dockerd-current
20743 root      20   0  207272   3576   1536 S   0.0  0.1   0:00.03 docker-containe
[vmindru@vmutil docker-cp]$ 
```

expected behavior, docker will perform data transfer in some sort of streamed manner and not consumed that amount of memory

**Comment by gianarb** [comment]
Hello thanks for your report
can you paste your `docker info` please?

**Comment by vmindru** [comment]
sure 

```
[vmindru@vmutil ~]$ docker info
Containers: 5
 Running: 1
 Paused: 0
 Stopped: 4
Images: 1
Server Version: 1.12.5
Storage Driver: devicemapper
 Pool Name: docker-253:2-132586-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 2.687 GB
 Data Space Total: 107.4 GB
 Data Space Available: 642.8 MB
 Metadata Space Used: 2.175 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 642.8 MB
 Thin Pool Minimum Free Space: 10.74 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.135-RHEL7 (2016-09-28)
Logging Driver: journald
Cgroup Driver: systemd
Plugins:
 Volume: local
 Network: bridge host null overlay
 Authorization: rhel-push-plugin
Swarm: inactive
Runtimes: docker-runc runc
Default Runtime: docker-runc
Security Options: seccomp selinux
Kernel Version: 3.10.0-514.2.2.el7.x86_64
Operating System: Red Hat Enterprise Linux Server 7.3 (Maipo)
OSType: linux
Architecture: x86_64
Number of Docker Hooks: 2
CPUs: 2
Total Memory: 2.779 GiB
Name: vmutil
ID: ZM37:KJ57:7WDA:A66F:2NRQ:QGX3:JQDO:X2GP:O6UH:5O3O:EZZA:KJEP
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://registry/v1/
Insecure Registries:
 127.0.0.0/8
Registries: registry(secure), docker.io (secure)
[vmindru@vmutil ~]$ 
```

**Comment by thaJeztah** [comment]
Let me close this ticket for now, as it looks like it went stale.
